ast/builtins.go

```diff
@@ -524,7 +525,7 @@ var CastObject = &Builtin{
 // RegexMatch takes two strings and evaluates to true if the string in the second
 // position matches the pattern in the first position.
 var RegexMatch = &Builtin{
-	Name: "re_match",
+	Name: "regex.match",
```

Comment:
This needs to be done incrementally, i.e., add another built-in function declaration called "regex.match" with the same signature and then put a comment on the old one saying it's deprecated, update the docs, etc. In a couple releases can remove the old from the codebase, although given the nature of re_match, I suspect it'll be around for a while.